### ISSUE TITLE
[docs] Use of private_key_file in octopusdeploy_ssh_key_account needs to be updated

### ISSUE THREAD

**Issue by jbristowe** [OctopusDeployLabs/terraform-provider-octopusdeploy#343]
Recently, a customer posed the following question concerning the `octopusdeploy_ssh_key_account` resource:

> What is the way to configure the key file? I tried all possible way like `./key`, `[keyfile]`, `[./keyfile]`.
> 
> ```terraform
> resource "octopusdeploy_ssh_key_account" "sshaccount" {
>   name             = "SSH Key Pair Account"
>   private_key_file = "example.key"
>   username         = "example"
> }
> ```

The [associated documentation for the resource is out-of-date](https://registry.terraform.io/providers/OctopusDeployLabs/octopusdeploy/latest/docs/resources/ssh_key_account) and needs to be updated; the property, `private_key_file` needs to be the Base64-encoded private key; not a path to the key file itself.

**Comment by Clare-Octopus** [comment]
Another instance of this happening (internal link) - https://octopus.zendesk.com/agent/tickets/100711

**Comment by Clare-Octopus** [comment]
For those coming across this issue - we could not get it working with putting the private key in full base-64 encoded text but we could get it working by putting a generic Private Key Passphrase in (we had a PEM file with no passphrase so you can either do that and put 'password' in like we have or if your key has a passphrase you will need to put that in):

```
resource "octopusdeploy_ssh_key_account" "example" {
 name            = "aws-linux-test-ssh-account"
 private_key_file = "${file("C:\\temp\\PEM.txt")}"
 private_key_passphrase ="password"
 username        = "octoadmin"
 environments = ["environments-22"]
 provider = octopusdeploy.space_terraform

}
```

**Comment by Clare-Octopus** [comment]
For those reading this - there is an issue at the moment with [the code ](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/3479efcf7a632ebf28e5246164a76cea97dcb890/octopusdeploy/schema_ssh_key_account.go#L16) as it currently references `private_key_passphrase` for the Key File.

If you want to get this to work you are going to need to put the base 64 encoded text into the value of `private_key_passphrase`, you still need to have a value for `private_key_file` but you can set this to any text you want. An example of this would be:

```
resource "octopusdeploy_ssh_key_account" "example" {
 name            = "aws-linux-test-ssh-account"
 private_key_file = "Randomtext"
 private_key_passphrase ="Ci0tLS0tQkVHSU4gT1BFTlNTSCBQUklWQVRFIEtFWS0tLS0tCmIzQmxibk56YUMxclpYa3RkakVBQUFBQUJHNXZibVVBQUFBRWJtOXVaUUFBQUFBQUFBQUJBQUFCbHdBQUFBZHpjMmd0Y24KTmhBQUFBQXdFQUFRQUFBWUVBekVWSEdiT3o5YkNGTzVtcVJZWGEyZGdxQVJ1aWZ5amYwVnlRMUpKRmNwcXQzUDd5RE5IZwpEblF0aUpvOWVERDNLZGtGQUxORHBNVVdpNTgya0pkRWVvbDJ4aFF4dEd1d083MEt3ZmJyZXN4TUhia3ZmSjVsMG1RRDRqClNicWZhY01XNkZONDlta3M5WHZsV2xyUG5td05yRjJuV09hcGR2dUVEUWtBbVBWdjdLenMxamoxcHfjjrngrngrgjnfjemfjtbnjfnrfjngjbnfjivnjrm9erfVmp1V1dqYVJxYlZWanh1cldNQ2NON1daTkRINTZNZy9EWVRjd2dBQXdhcHNleGZlZmsza3RXQjlhbFQ3MXhxMDZRClRobWU3NjlibzZmYjA0di8zZktmZFYxcDFCZm9WblF5V1VwWE4wUTJVRjdqMk4ySWkrfrkmvgkbmgpobk,rpvbmkmgftkmbokmfkdfopgmJDKGNEWOGMKFKBbkRlMW1UUXgrZWpJUHcyRTNNSUFBTUdxYkhzWDNuNU41TFZnZldwVSs5Y2F0T2tFNFpudSt2VzZPbgoyOU9MLzkzeW4zVmRhZFFYNkZaME1sbEtWemRFTmxCZTQ5amRpSXZveU9lQ0g5QWNXaWllWmFCeEsxY3ZZajJadjhzSXgwCkFPSDdZUERRZ0Ivd0ZHVVA1QiswUmhydkNmSmpzblkzVU1lSlpGUklBZHdTNUN3UUoyRHpwWUhES294NnNRdXpwdEV2UXMKYlFSQndIWnZvZU9HV3Mya0lMYUE1d3hFN0FBUjhRQUFBQU1CQUFFQUFBR0FBM3FYdGRRMHFmcGlFVncyd0paYzNqM0dwZQo1MzF3RzE2ckVIVVVkNnZVN0l6QnJmY0x3cTROaTE5NGJsaFdSY3FtV2xGRTlDYWM0ZGZGTDRqSC9ibG8vT3hadjdTd042rkornfrnrjfmgbognjrnfjgvnbjnrjfngvjnbjnrejnfvjfgnemfnbjgnvgretXUmo4WFZXQVVBQUFEQkFOeis5WTZDbG0vcjREN00KWXA2NlU5Z2RVUVhTZk5kNlZjcyt6UFNUTkFQY3ZtY2dSaHNTOU9mQ1dCdDJIMG1DOWtiZFNqSjgwN2pJVFlQdlpGbmtPLwphd3daNzk4Tk5leHBZWlBHVHIyTzNoenQ3WkthSTFhbllDQW1pYXk5UXZyQzJSNnlsSWNTTHE1VFJRMGhCeUp6TjRReWxzCmlxaThPai9rUGd1Z0xwVkdDMGtYTmFUT09DTzdYOXduUDZFSmxaZmRTS0pRTEdDUFR2RWN3WXExSk5TZzlUOVhsa2tJck0KTzgyOVoxREpSWGNoSzF4TmZaSDhhZ3Fyc2E3YzRzeXdBQUFNRUE3S0FpQ0JrWnF6bkJoRXJySXhwQ3JKUzlyTEdibXBpSgpLTm5oRWZPSWcwN0pMMDVJbllJQTY4OHhaSkdxL0FFTjFvLzdVSE9RMVl6S1dEZFYyODlmcUdzKzdCa0dUVGJZNnJkRTVnCkw1TlBrb1pSOU9VTWdseUx1Sy9PTE1GTzNaUjJRZS84aUFGRGU0V0t2bmxwUE54ejRMV3RobGhWektKNDJyUEtueFY1aEcKL04zNmdVUHVBNGxCWFpiNC9qVFFoQ3pOWEZDSHlJald0ZlY5TzF1UmNJTWlZOTJsQlVGSkZ6bTcybUxCZGtIbnBCV21oaQpFcHZiUTI3OVFjYkVDekFBQUFDMnAzWVd4emFFQkJhVzl1QVFJREJBVUdCdz09Ci0tLS0tRU5EIE9QRU5TU0ggUFJJVkFURSBLRVktLS0tLQo="
 username        = "octoadmin"
 environments = ["environments-22"]
 provider = octopusdeploy.space_terraform
```

Best practise would be to put the base 64 text into a sensitive variable in Octopus and reference that variable in your terraform file:

```
resource "octopusdeploy_ssh_key_account" "example" {
 name            = "aws-linux-test-ssh-account"
 private_key_file = "Randomtext"
 private_key_passphrase =#{KEY_FILE}
 username        = "octoadmin"
 environments = ["environments-22"]
 provider = octopusdeploy.space_terraform
```

**Comment by EduardoSantanaSeverino** [comment]
Thank you @Clare-Octopus  

We are planning to use this functionality into our CI/CD. 

But seeing this Issue, open since 2022 makes me double think about it. 

Probably there is not support available for this Terraform Provider.

**Comment by Clare-Octopus** [comment]
Good morning @EduardoSantanaSeverino,

Thank you for reaching out, our Terraform provider is still in active development as you can see from our documentation [here](https://registry.terraform.io/providers/OctopusDeployLabs/octopusdeploy/latest/docs). 

I will mention this GitHub issue to our engineers and see if they can bump it up on their list of issues to look at but since this feature is still under active development it may still be awhile before they can look into it.

Kind Regards,
Clare

**Comment by NickJosevski** [comment]
@EduardoSantanaSeverino thanks for your patience, it's not practical for us to fix every issue that comes up 😁 

So the more that people comment on any given issue that helps give signal as to what we should prioritise / consider.

**Comment by EduardoSantanaSeverino** [comment]
@Clare-Octopus 

Thank you for the clarification.

**Comment by zentron** [comment]
This issue has been resolves as part of a breaking change in https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/642. and shipped in `v0.21.1`

**Comment by EduardoSantanaSeverino** [comment]
thank you. I see the new version. and it was nice to have this update. thanks to everyone involved.
